syntax = 'proto2';

import "Computer.proto";
import "Shop.proto";
import "Item.proto";

option optimize_for = LITE_RUNTIME;
package GSToGC;

enum MsgId {
    eUserGameInfo = 10;
    eUserPayInfo = 11;
    eAskRet = 12;
    eSyn_ShopPriceList = 13;
    eSyn_Inventory = 14;
}

message UserGameInfo {
    optional MsgId msgid = 1 [default = eUserGameInfo];
    required uint32 gold = 2;
    required uint32 energy = 3;
    repeated ComputerInfo comp = 4;

}

message SynShopPriceList {
    optional MsgId msgid = 1 [default = eSyn_ShopPriceList];
    repeated PageShop page = 2;
}

message SynInventory {
    optional MsgId msgid = 1 [default = eSyn_Inventory];
    required ItemInfo item = 2;
}

message UserPayInfo {
    optional MsgId msgid = 1 [default = eUserPayInfo];
    required uint32 gold = 2;
    required uint32 energy = 3;

}

message AskRet {
    optional MsgId msgid = 1 [default = eAskRet];
    required uint32 askid = 2;
    required uint32 errorcode = 3;
}